### PR TITLE
[BPF] host can access self via a service without CTLB

### PR DESCRIPTION
**Pull request by tomastigera**
added test for this case

No host source conflict resolution on lo since we cannot tell traffic straight to self from traffic to self via a service. Host connecting to self and connecting to the same self port via a service should be rare, but if that is necessaty, than CTLB is the only option.

fixes https://github.com/projectcalico/calico/issues/8557
## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
ebpf: host can access self via a service without CTLB
```

## Reminder for the reviewer

Make sure that this PR has the correct labels and milestone set.

Every PR needs one `docs-*` label.

- `docs-pr-required`: This change requires a change to the documentation that has not been completed yet.
- `docs-completed`: This change has all necessary documentation completed.
- `docs-not-required`: This change has no user-facing impact and requires no docs.

Every PR needs one `release-note-*` label.

- `release-note-required`: This PR has user-facing changes. Most PRs should have this label.
- `release-note-not-required`: This PR has no user-facing changes.

Other optional labels:

- `cherry-pick-candidate`: This PR should be cherry-picked to an earlier release. For bug fixes only.
- `needs-operator-pr`: This PR is related to install and requires a corresponding change to the operator.

### CODE REVIEW

File: felix/bpf-gpl/tc.c
Comment (fasaxc) [inline felix/bpf-gpl/tc.c:272]: Typo: loopi

**Review by fasaxc — APPROVED** [review]
One typo but LGTM

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and leave the branch after I've merged it.

**Pull request merged**
